### PR TITLE
[4.0] validation message number field

### PR DESCRIPTION
**Pull request by brianteeman**
PR for #30397 

Currently if you set the value for a number field outside of the min/max value then the form will save BUT the value you entered is not saved and instead the default value is saved.

This PR is a DRAFT and help is requested
So far you will get an error message when you save but the form is still saved.
~The error message does include what the minimum or maximum value is but it doesn't include the name of the field~

image

### CODE REVIEW

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on 85ec29e17c69a7a264e79ec29a4440b44e16eac3<br /><br />Test done with the value -1 at the field Session Lifetime (minutes) (Global Configuration -> System).
Done!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30906">issues.joomla.org/tracker/joomla-cms/30906</a>.</sub>

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 85ec29e17c69a7a264e79ec29a4440b44e16eac3<br /><br />A good place to test this is in the User Options form - Password Options tab. I set two out of range and got a message for both of them. So the residual problem is that a valid number, say Minimum Characters set to 8, is set to the default value of 12 because I set Maximum Reset Count to 50, out of range.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30906">issues.joomla.org/tracker/joomla-cms/30906</a>.</sub>

**Comment by brianteeman** [comment]
Thanks for testing but this is not complete because although you get the error message the form is still saved

**Comment by brianteeman** [comment]
Anyone able to help me get this finished

**Comment by gostn** [comment]
Test done with the value -1 at the field Session Lifetime. With pr got red and yellow warning, without pr got yellow one. In both cases the default value is saved. As i dont really understand the instructions i wanna ask if this test is successfull.

image

**Comment by brianteeman** [comment]
@gostn follow the steps to reproduce in the original issue #30397

**Comment by gostn** [comment]
> follow the steps to reproduce in the original issue #30397

not sure what author mean, dont understand.

File: libraries/src/Form/Rule/NumberRule.php [hunk 1]
```diff
@@ -55,6 +57,8 @@ public function test(\SimpleXMLElement $element, $value, $group = null, Registry
 
 			if ($min > $float_value)
 			{
+				Factory::getApplication()->enqueueMessage(Text::plural('JFIELD_NUMBER_TOO_SHORT_N', $min, Text::_($element['label'])), 'error');
```
Comment (SharkyKZ) [inline libraries/src/Form/Rule/NumberRule.php:60]: Change this to throw a `UnexpectedValueException`. This avoids the duplicated warning.

Comment (brianteeman) [reply libraries/src/Form/Rule/NumberRule.php:60]: could you expand on that suggestion with the full syntax please

Comment (Fedik) [reply libraries/src/Form/Rule/NumberRule.php:60]:
```php
throw new \UnexpectedValueException(Text::plural('JFIELD_NUMBER_TOO_SHORT_N', $min, Text::_($element['label'])));
```
same for "too long"

Comment (brianteeman) [reply libraries/src/Form/Rule/NumberRule.php:60]: thats what I tried and it just putputs the success message

Comment (Fedik) [reply libraries/src/Form/Rule/NumberRule.php:60]: hm, then there a more bugs, somewhere else

Comment (Fedik) [reply libraries/src/Form/Rule/NumberRule.php:60]: I not tested but I suspect this should work after  pr #31105

**Pull request closed without merging**